### PR TITLE
Send MemberIsUnavailable on executor thread

### PR DESCRIPTION
**Pull request by rickardoberg**
MemberIsAvailable/MemberIsUnavailable messages could arrive out of order if an instance is switching to master and another instance is elected as master (first unavailable, then available). This fixes so that messages are sent on the state switching thread, so the order is correct.

Also, ProtocolServer.msgLog was a GC root, causing OOMs. Clear on shutdown to fix.

Do not forward merge this!

### CODE REVIEW

File: enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java [hunk 1]
```diff
@@ -367,14 +366,18 @@ void checkDataConsistency( MasterClient masterClient, NeoStoreDataSource neoData
...
 
-    private void checkMyStoreIdAndMastersStoreId( NeoStoreDataSource neoDataSource, boolean masterIsOld )
+    private void checkMyStoreIdAndMastersStoreId( NeoStoreDataSource neoDataSource, boolean masterIsOld, URI masterUri )
     {
         if ( !masterIsOld )
         {
             StoreId myStoreId = neoDataSource.getStoreId();
 
             ClusterMembers clusterMembers = resolver.resolveDependency( ClusterMembers.class );
-            ClusterMember master = single( filter( inRole( MASTER ), clusterMembers.getMembers() ) );
```
Comment (digitalstain) [inline enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java:383]: It would be useful to explain in the commit message why this change is necessary. Also, this appears to be breaking the PR build.

**Comment by davidegrohmann** [comment]
@rickardoberg needs rebase
